interface/src/lib.rs

```diff
@@ -195,6 +195,15 @@ impl From<[u8; 32]> for Txid {
     }
 }
 
+impl From<Vec<u8>> for Txid {
+    fn from(vec: Vec<u8>) -> Self {
+        let bytes: [u8; 32] = vec
+            .try_into()
+            .expect("Can't convert into Txid, vector must be 32 bytes");
+        Self(bytes)
+    }
+}
```

Comment:
Today I learned about `&*`.

In the [MR13407](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/13407) there were cases where I had to convert vectors into `Txid`. Before this MR there were slices that got converted to `Txid` just by using `.into()`. There were no such thing for a vector, so I worked that around with this

```rust
fn vec_to_txid(vec: Vec<u8>) -> Txid {
    let bytes: [u8; 32] = vec.try_into().expect("Can't convert to [u8; 32]");
    bytes.into()
}
``` 

Which is ugly, because I had to repeat it in every file, or share it somehow. The most appropriate seemed to me was to just add an `.into()` support for vectors.

But "just" in Rust is not simply "just". I can't re-implement `TryFrom` for vectors because it is in conflict with a blanket implementation. So I used a `From` only with a panic, which is not great, I know.

The other alternative is not to add anything for vectors, but to use `&*` to dereference a vector and then to use it as a reference to a slice.

```rust
let src = vec![1u8; 32];
let result: Txid = (&*src).try_into().unwrap();
```

I think I will just close this PR, since adding a proper `TryFrom` seems as too much of a trouble for too little of a win.